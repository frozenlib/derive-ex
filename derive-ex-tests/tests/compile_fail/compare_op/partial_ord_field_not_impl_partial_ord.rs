use derive_ex::derive_ex;

#[derive(PartialEq)]
struct NotPartialOrd;

#[derive_ex(PartialOrd, PartialEq)]
struct X {
    x: NotPartialOrd,
}

fn main() {}
