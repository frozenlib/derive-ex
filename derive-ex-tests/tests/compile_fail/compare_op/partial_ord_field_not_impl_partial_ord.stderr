error[E0277]: can't compare `NotPartialOrd` with `NotPartialOrd`
 --> tests/compile_fail/compare_op/partial_ord_field_not_impl_partial_ord.rs:8:5
  |
8 |     x: NotPartialOrd,
  |     ^ no implementation for `NotPartialOrd < NotPartialOrd` and `NotPartialOrd > NotPartialOrd`
  |
  = help: the trait `PartialOrd` is not implemented for `NotPartialOrd`
help: consider annotating `NotPartialOrd` with `#[derive(PartialOrd)]`
  |
4 + #[derive(PartialOrd)]
5 | struct NotPartialOrd;
  |
